### PR TITLE
Support PolicyReport CRD from the Kubernetes Policy WG

### PR DESCRIPTION
**Pull request by fjogeleit**
PolicyReport and ClusterPolicyReport are CRDs created by the Kubernetes Policy WG (https://github.com/kubernetes-sigs/wg-policy-prototypes). The goal is to create a default Schema for Policy related tools like Kyverno, Falco, Kube Bench and other tools like JsPolicy. This makes it possible to create generell Tooling for this Projects like [Policy Reporter](https://github.com/kyverno/policy-reporter).

This PR adds this CRDs and create them besides the existing JsPolicyViolations CRD. It creates one PolicyReport per Namespace for all JsPolices and one ClusterPolicyReport for all Cluster scoped violations.

This makes it possible to use JsPolicy together with Policy Reporter and creates observability capabilities like integrations in Prometheus, Grafana Loki or the standalone Policy Reporter UI

terminal

policy-reporter-ui

prometheus-metrics

grafana-loki

### CODE REVIEW

File: pkg/webhook/webhook.go [hunk 1]
```diff
@@ -120,6 +121,7 @@ func (wh *Webhook) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 	if reviewResponse.Allowed == false {
 		if jsPolicy.Spec.AuditPolicy == nil || *jsPolicy.Spec.AuditPolicy != policyv1beta1.AuditPolicySkip {
 			go LogRequest(context.TODO(), wh.Client, req, reviewResponse, jsPolicy, wh.Scheme, 5)
+			go ReportRequest(context.TODO(), wh.Client, req, reviewResponse, jsPolicy, wh.Scheme, 5)
```
Comment (FabianKramm) [inline pkg/webhook/webhook.go:124]: please create an environment variable `ENABLE_POLICY_REPORTS` that needs to be `true` for this to get executed

**Review by FabianKramm — CHANGES_REQUESTED** [review]
@fjogeleit thanks for this PR! Sorry for the delay, in general the PR looks good, however I guess it would be better for now to have some sort of feature flag (like an environment variable) that enables this reporting type instead of the current `JsPolicyViolations` reporting as it seems like overlapping functionality to me that should not by default be enabled together. In future, we could think about making this the default reporting type.

File: pkg/webhook/webhook.go (hunk 1 above)
Comment (fjogeleit) [reply pkg/webhook/webhook.go]: I added this ENV variable and a second one `POLICY_REPORT_MAX_EVENTS` to configure the max amount of events per report. I also mode this env vars configurable over the helm chart

**Comment by FabianKramm** [comment]
@fjogeleit thanks a lot for the PR! Looks good to me now!

**Pull request merged**